scripts/natural_language_inference/main_intra.py

```diff
@@ -0,0 +1,146 @@
+import mxnet as mx
+from mxnet import gluon, autograd, nd
+from decomposable_atten import DecomposableAtten, IntraSentenceAtten
+from nlidataset import NLIDataset, NLIDataItem
+import gluonnlp as nlp
+import argparse
+import os
+from utils import *
+import numpy as np
+
+label_to_idx = {'neutral': 0, 'contradiction': 1, 'entailment': 2}
+
+class Network(gluon.Block):
+    def __init__(self, word_embed_size, hidden_size, **kwargs):
+        super(Network, self).__init__(**kwargs)
+        self.word_embed_size = word_embed_size
+        self.hidden_size = hidden_size
+        with self.name_scope():
+            self.lin_proj = gluon.nn.Dense(hidden_size, in_units=word_embed_size, activation="relu")
+            self.intra_atten = IntraSentenceAtten(hidden_size, hidden_size)
+            self.model = DecomposableAtten(hidden_size*2, hidden_size, 3)
+
+    def forward(self, sentence1, sentence2):
+        B, L1, D = sentence1.shape
+        B, L2, D = sentence2.shape
+        s1 = self.lin_proj(sentence1.reshape(B * L1, D)).reshape(B, L1, self.hidden_size)
+        s1 = mx.nd.concat(s1, self.intra_atten(s1), dim=-1)
+        s2 = self.lin_proj(sentence2.reshape(B * L2, D)).reshape(B, L2, self.hidden_size)
+        s2 = mx.nd.concat(s2, self.intra_atten(s2), dim=-1)
+        pred = self.model(s1, s2)
+        return pred
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--data_root', help="root of data file", default="./data")
+    parser.add_argument('--train_set', help="training set file", default="snli_1.0/snli_1.0_train.txt")
+    parser.add_argument('--dev_set', help="development set file", default="snli_1.0/snli_1.0_dev.txt")
+    parser.add_argument('--batch_size', help="batch_size", default=32, type=int)
+    parser.add_argument('--print_period', help="the interval of two print", default=20, type=int)
+    parser.add_argument('--checkpoints', help="path to save checkpoints", default="checkpoints")
+    parser.add_argument('--model', help="model file to test, only for test mode", default=None)
+    parser.add_argument('--mode', help="train or test", default="train")
+    return parser.parse_args()
+
+def train_network(model, train_set, embedding, ctx, args):
+    model.collect_params().initialize(mx.init.Xavier(), force_reinit=True, ctx=ctx)
```

Comment:
I found that it is impossible for the model to converge with such initialization. It seems that initializing from Gaussian with mean 0 and standard deviation 0.01 will cause the model stuck at a point that accuracy is around 1/3.

Reply:
I am still tuning it. Using `mx.init.MSRAPrelu()` leads to fastest convergence, and it is easy to explain since that this initialization is proposed for ReLU. `Xavier()` also works. But the initialization method proposed in the paper does not work at all.